### DISCUSSION TITLE
LocalAI embeddings

### DISCUSSION THREAD

**Discussion by etlweather** [logancyang/obsidian-copilot#738] [General]
I am using [Local AI](https://github.com/mudler/LocalAI) as my local AI engine. I set up Obsidian to use a "3rd party (openai-format)" model for embeddings.

If I use `curl`, I can get embeddings for text from my Local AI server with the same model name I provided.

But when I try to index my vault, all I get is errors on the Local AI backend and in Obsidian. The error in the console isn't useful (even in debug mode) from what I can tell, all it tells me is that it got a 500 error back and I have debug turned on.

What am I doing wrong?

**Comment by logancyang** [comment]
You should check the server log for 500 errors. If you are not comfortable with that, I suggest other easier alternatives such as Ollama.

**Reply by etlweather** [reply]
I can see the error on LocalAI and that isn't particularly useful because it does not tell me what was sent by Copilot and Copilot's errors does not show me the request either.

**Reply by etlweather** [reply]
It looks like the request isn't properly formed.

**Reply by logancyang** [reply]
@etlweather The thing is for users who choose to use LocalAI they need to be a bit technical. The entire code of Copilot is open-source, and localAI has server log. A technical user can debug by themselves with these. My recommendation for users who are less technical is to choose Ollama or LM Studio which are listed as explicitly supported provider options.

**Reply by etlweather** [reply]
Well I can spend time debugging it, but one thing I would like to ask - is there a way to see the network requests made by the Copilot plugin in the DevTools? I opened the DevTools but it shows no network requests.

**Reply by etlweather** [reply]
Appears the problem is Obsidian Copilot sends an array of strings in the request and LocalAI does not seem to support it. As this is valid OpenAI format, I am opening an issue with LocalAI.

**Reply by etlweather** [reply]
Hum, nope, that's not the problem.

**Reply by logancyang** [reply]
@etlweather what's the embedding model you use and what's its context length? I'm not sure how localAI embedding models handle overly long chunks, whether it can trigger errors or fail silently (like ollama).

**Reply by etlweather** [reply]
Yeah, I thought it might be the length of the text too. I managed to extract a failing request and it has 4,480 chars. I am using `bert-all-MiniLM-L6-v2` The answer as to what happens on too large context, it responds with an error (status 500).
